### PR TITLE
[ready] `scale`: kokkos impl

### PR DESCRIPTION
**Pull request by fnrizzi**
#162  is a prerequisite for this, I can rebase once that is merged

### CODE REVIEW

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp [hunk 1]
```diff
@@ -3,17 +3,35 @@
...
+         std::experimental::default_accessor<ElementType>
+       > obj)
 {
-    KokkosBlas::scal(Impl::mdspan_to_view(x),alpha,Impl::mdspan_to_view(x));
-}
+  // constraints
+  static_assert(obj.rank() <= 2);
+
+#if defined KOKKOS_STDBLAS_ENABLE_TESTS
+  std::cout << "scale: kokkos impl\n";
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp:29]: Please see my comment on e.g., PR #160 about encapsulating logging.  This is not required but would be a good idea (I think : - ) ).  Thanks!

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp [hunk 2]
```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
 };
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
+std::enable_if_t<
+  MDSpan::rank()==1
+  && !is_complex_v<typename MDSpan::element_type>,
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp:35]: Please see my comments on PR #162.  This won't work for e.g., `mdspan<const std::complex<T>, ...>`.

**Review by mhoemmen — CHANGES_REQUESTED** [review]

**Comment by fnrizzi** [comment]
@mhoemmen this is ready IMO

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp [hunk 3]
```diff
@@ -2,18 +2,34 @@
...
+template<class ExeSpace,
          class Scalar,
          class ElementType,
          std::experimental::extents<>::size_type ... ext,
-         class Layout,
-         class Accessor>
-void scale(kokkos_exec<ExecSpace>, const Scalar alpha,
-           std::experimental::mdspan<ElementType, std::experimental::extents<ext ...>, Layout, Accessor> x)
+         class Layout>
+requires (sizeof...(ext) <= 2)
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp:18]: This is a perfectly fine way to express the constraint.  You could also add a `requires` clause after the function's parameters, and do `requires(obj.rank() <= 2)`.

**Review by mhoemmen — APPROVED** [review]
Looks great!  Thanks for the contribution and for your responses to feedback!

It looks like there's a merge conflict, alas.

**Comment by fnrizzi** [comment]
@mhoemmen conflict fixed via #172

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp (hunk 3 above)
Comment (mhoemmen) [reply tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_scale_kk.hpp:18]: Just to clarify: Either way is fine with me : - )

**Review by mhoemmen — APPROVED** [review]

**Pull request merged**